src/sa/internal/sa_interface_inmemory.template.c

```diff
@@ -1306,58 +1309,66 @@ static int32_t sa_create(void)
+        temp_sa->shplf_len  = ((uint8_t)sdls_frame.pdu.data[3] & 0x03);
+        temp_sa->stmacf_len = ((uint8_t)sdls_frame.pdu.data[4]);
+        temp_sa->ecs_len    = ((uint8_t)sdls_frame.pdu.data[5]);
+        for (x = 0; x < temp_sa->ecs_len; x++)
         {
-            sa[spi].ecs = ((uint8_t)sdls_frame.pdu.data[count++]);
+            temp_sa->ecs = ((uint8_t)sdls_frame.pdu.data[count++]);
         }
-        sa[spi].shivf_len = ((uint8_t)sdls_frame.pdu.data[count++]);
-        for (x = 0; x < sa[spi].shivf_len; x++)
+        temp_sa->shivf_len = ((uint8_t)sdls_frame.pdu.data[count++]);
+        for (x = 0; x < temp_sa->shivf_len; x++)
         {
-            sa[spi].iv[x] = ((uint8_t)sdls_frame.pdu.data[count++]);
+            temp_sa->iv[x] = ((uint8_t)sdls_frame.pdu.data[count++]);
         }
-        sa[spi].acs_len = ((uint8_t)sdls_frame.pdu.data[count++]);
-        for (x = 0; x < sa[spi].acs_len; x++)
+        temp_sa->acs_len = ((uint8_t)sdls_frame.pdu.data[count++]);
+        for (x = 0; x < temp_sa->acs_len; x++)
         {
-            sa[spi].acs = ((uint8_t)sdls_frame.pdu.data[count++]);
+            temp_sa->acs = ((uint8_t)sdls_frame.pdu.data[count++]);
         }
-        sa[spi].abm_len = (uint8_t)((sdls_frame.pdu.data[count] << BYTE_LEN) | (sdls_frame.pdu.data[count + 1]));
+        temp_sa->abm_len = (uint8_t)((sdls_frame.pdu.data[count] << BYTE_LEN) | (sdls_frame.pdu.data[count + 1]));
         count           = count + 2;
-        for (x = 0; x < sa[spi].abm_len; x++)
+        for (x = 0; x < temp_sa->abm_len; x++)
         {
-            sa[spi].abm[x] = ((uint8_t)sdls_frame.pdu.data[count++]);
+            temp_sa->abm[x] = ((uint8_t)sdls_frame.pdu.data[count++]);
         }
-        sa[spi].arsn_len = ((uint8_t)sdls_frame.pdu.data[count++]);
-        for (x = 0; x < sa[spi].arsn_len; x++)
+        temp_sa->arsn_len = ((uint8_t)sdls_frame.pdu.data[count++]);
+        for (x = 0; x < temp_sa->arsn_len; x++)
         {
-            *(sa[spi].arsn + x) = ((uint8_t)sdls_frame.pdu.data[count++]);
+            *(temp_sa->arsn + x) = ((uint8_t)sdls_frame.pdu.data[count++]);
         }
-        sa[spi].arsnw_len = ((uint8_t)sdls_frame.pdu.data[count++]);
-        for (x = 0; x < sa[spi].arsnw_len; x++)
+        temp_sa->arsnw_len = ((uint8_t)sdls_frame.pdu.data[count++]);
+        for (x = 0; x < temp_sa->arsnw_len; x++)
         {
-            sa[spi].arsnw = sa[spi].arsnw | (((uint8_t)sdls_frame.pdu.data[count++]) << (sa[spi].arsnw_len - x));
+            temp_sa->arsnw = temp_sa->arsnw | (((uint8_t)sdls_frame.pdu.data[count++]) << (temp_sa->arsnw_len - x));
         }
 
-        // TODO: Checks for valid data
-
         // Set state to unkeyed
-        sa[spi].sa_state = SA_UNKEYED;
+        temp_sa->sa_state = SA_UNKEYED;
+
+        // Verify data
+        status = sa_verify_data(temp_sa);
+
+        if (status == CRYPTO_LIB_SUCCESS)
+        {
+            sa[spi] = *temp_sa;
```

Comment:
Is this correct?

Reply:
You're right, I think we'd need to do a deep copy here as temp_sa will be freed as this function exits